targets/TARGET_STM/hal_tick_common.c

```diff
@@ -17,9 +17,19 @@
 
 // Overwrite default HAL functions defined as "weak"
 
+// This variable is set to 1 at the of mbed_sdk_init function.
+// The ticker_read_us function must not be called until the mbed_sdk_init is terminated.
+extern int mbed_sdk_inited;
+
 uint32_t HAL_GetTick()
 {
-    return ticker_read_us(get_us_ticker_data()) / 1000; // 1 ms tick is required for ST HAL
+    // 1 ms tick is required for ST HAL driver
+    if (mbed_sdk_inited) {
+        return (ticker_read_us(get_us_ticker_data()) / 1000);
```

Comment:
This is going to start counting from 0 so there will likely be a time jump after the SDK is initialized. As a workaround you may want to record the time returned by `us_ticker_read` just before mbed_sdk_inited is set and use it as an offset.

Reply:
Done in commit https://github.com/ARMmbed/mbed-os/pull/7352/commits/f785c23e89d1a5adb456cdd8da10dc06e24bd22c